shiny/ui/_theme.py

```diff
@@ -0,0 +1,513 @@
+from __future__ import annotations
+
+import copy
+import os
+import pathlib
+import tempfile
+from textwrap import dedent
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    Literal,
+    Optional,
+    Sequence,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from htmltools import HTMLDependency
+from packaging.version import Version
+
+from .._docstring import no_example
+from .._typing_extensions import NotRequired, TypedDict
+from .._versions import bootstrap
+from ._theme_presets import (
+    ShinyThemePreset,
+    shiny_theme_presets,
+    shiny_theme_presets_bundled,
+)
+from ._utils import path_pkg_www
+
+T = TypeVar("T", bound="Theme")
+
+
+SassImporterReturnValue = Union["tuple[str]", "tuple[str, str]", "tuple[str, str, str]"]
+SassImporterFunction = Union[
+    Callable[[str], SassImporterReturnValue],
+    Callable[[str, str], SassImporterReturnValue],
+]
+
+
+class SassCompileArgs(TypedDict):
+    output_style: NotRequired[Literal["nested", "expanded", "compact", "compressed"]]
+    source_comments: NotRequired[bool]
+    source_map_contents: NotRequired[bool]
+    source_map_embed: NotRequired[bool]
+    omit_source_map_url: NotRequired[bool]
+    source_map_root: NotRequired[str | None]
+    include_paths: NotRequired[Sequence[str]]
+    precision: NotRequired[int]
+    custom_functions: NotRequired[Any]  # not worth the effort, it's a complicated type
+    indented: NotRequired[bool]
+    importers: NotRequired[Iterable[tuple[int, SassImporterFunction]] | None]
+
+
+@no_example()
```

Comment:
Seems like it'd be nice to have at least one example on the reference page?

Also, more generally, are you thinking we avoid investing in other documentation until we have the more complete story?

Reply:
I can add an example. I was waiting to invest time in documentation until I had at least one review and an indication that we weren't going to radically change the implementation.